src/business-logic/services/device-service-implementation.ts

```diff
@@ -0,0 +1,80 @@
+import { DeviceService } from './interfaces/device-service'
+import { MongoDeviceRepository } from '../../data-access/repositories/mongo/mongo-device-repository'
+import { Device } from '../../model/entities/device'
+import { INewsDevice } from '../../model/entities/inews-device'
+import { TelemetricsDevice } from '../../model/entities/telemetrics-device'
+
+export class DeviceServiceImplementation implements DeviceService {
+  constructor(private readonly deviceRepository: MongoDeviceRepository) {}
+  
+  public async readAllConfigurations(): Promise<(Device | INewsDevice | TelemetricsDevice)[]> {
+    return await this.deviceRepository.findAllDevices()
+  }
+
+  public async readConfiguration(deviceId: string): Promise<(Device | INewsDevice | TelemetricsDevice)> {
+    const deviceConfig = await this.deviceRepository.findById(deviceId)
+    if (!deviceConfig) {
+      throw new Error(`Device with id ${deviceId} not found`)
+    }
+    return deviceConfig
+  }
+
+  public async create(_device: Device | INewsDevice | TelemetricsDevice): Promise<void> {
+    const existingDevice = await this.deviceRepository.findById(_device.id)
+    if (existingDevice) {
+      throw new Error(`Device with id ${_device.id} already exists`)
+    }
+    await this.deviceRepository.create(_device)
+  }
+
+  public async update(deviceId: string, config: Device): Promise<void> {
+    const existingDevice = await this.deviceRepository.findById(deviceId)
+    if (!existingDevice) {
+      throw new Error(`Device with id ${deviceId} not found`)
+    }
+    await this.deviceRepository.update(deviceId, config)
+  }
+
+  public async delete(deviceId: string): Promise<void> {
+    const existingDevice = await this.deviceRepository.findById(deviceId)
+    if (!existingDevice) {
+      throw new Error(`Device with id ${deviceId} not found`)
+    }
+    await this.deviceRepository.delete(deviceId)
+  }
+
+  public async disconnect(deviceId: string): Promise<void> {
+    const deviceConfig = await this.deviceRepository.findById(deviceId)
+    if (!deviceConfig) {
+      throw new Error(`Device with id ${deviceId} not found`)
+    }
+    // Simulate disconnecting the device
+    console.log(`Disconnecting device ${deviceId}`)
+    await new Promise((resolve) => setTimeout(resolve, 1000))
+  }
+
+  public async connect(deviceId: string): Promise<void> {
+    const deviceConfig = await this.deviceRepository.findById(deviceId)
+    if (!deviceConfig) {
+      throw new Error(`Device with id ${deviceId} not found`)
+    }
+    // Simulate reconnecting the device
+    console.log(`Connecting to device ${deviceId}`)
+    await new Promise((resolve) => setTimeout(resolve, 1000))
+  }
+  
+  public async connectAll(): Promise<void> {
+    const devices = await this.deviceRepository.findAllDevices()
+
+    if (!devices) {
+      throw new Error('Devices not found')
+    }
```

Comment:
`findAllDevices` should never return undefined. If it doesn't have any Devices it should return an empty list.
That would make the type of `devices` to be `Device[]`, which would tell you that your check `if (!device)` isn't needed.
Just an example of the benefits of including types on the local variables :)